### PR TITLE
Minor fixes and updates

### PR DESCRIPTION
**Pull request by AdrianRaFo**
This PR includes minor fixes and clean up based on IntelliJ highlights. Also, update the project to Java 8

### CODE REVIEW

File: helios-benchmarks/build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 dependencies {
     compile project(":helios-core")
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlinVersion"
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
```
Comment (pakoito) [inline helios-benchmarks/build.gradle]: This prevents using it on Android AFAIK

File: helios-parser/src/main/kotlin/helios/parser/ByteBasedParser.kt [hunk 2]
```diff
@@ -30,7 +30,7 @@ interface ByteBasedParser<J> : Parser<J> {
         var j = i
         var c: Int = (byte(j) and 0xff.toByte()).toInt()
         while (c != 34) {
-            if (c < 32) return die(j, "control char ($c) in string")
+            if (c < 32) die(j, "control char ($c) in string")
```
Comment (pakoito) [inline helios-parser/src/main/kotlin/helios/parser/ByteBasedParser.kt]: This seems like a bug to me

Comment (AdrianRaFo) [reply helios-parser/src/main/kotlin/helios/parser/ByteBasedParser.kt]: Unreachable code following IntelliJ because of `die` returns `Nothing`

Comment (pakoito) [reply helios-parser/src/main/kotlin/helios/parser/ByteBasedParser.kt]: Literally an exception then

File: helios-parser/src/main/kotlin/helios/parser/AsyncParser.kt [hunk 3]
```diff
@@ -160,26 +157,25 @@ class AsyncParser<J>(
...
 
                         ',' -> {
-                            if (state == ASYNC_POSTVAL) {
-                                offset += 1
-                                state = ASYNC_PREVAL
-                            } else if (state == ASYNC_END) {
-                                die(offset, "expected eof")
-                            } else {
-                                die(offset, "expected json value")
+                            when (state) {
```
Comment (raulraja) [inline helios-parser/src/main/kotlin/helios/parser/AsyncParser.kt]: These chnages of `if` for `when` are not valid changes unless we can prove the bytecode emitted is more performant than with `if` statements. This parser is purposedly fast and internally mutable to get the best raw performance out of JSON parsing and speed comes before code style in this case. All other changes look good

Comment (AdrianRaFo) [reply helios-parser/src/main/kotlin/helios/parser/AsyncParser.kt]: I'll restore by now and check it after in a separate pr

**Pull request merged**